arduino/smartcar/smartcar.ino

```diff
@@ -155,7 +164,68 @@ void loop()
     
 }
 
-//If the connection breaks, this method will be called
+
+void stopZoneAutoBreak(long frontUltDis, long frontIRDis, long backIRDis)
+{
+     if (frontUltDis <= 100 && frontUltDis != 0 || frontIRDis <= 40 && frontIRDis != 0 || backIRDis <= 40 && backIRDis != 0)//stop zone
+         {
+            if (canDrive)//check whether you're in the stop zone
+            {
+                car.setSpeed(0);
+                Serial.println("Emergency stop 1");
+            }
+            canDrive = false;//so the car can move in the stop soon
```

Comment:
You wrote soon here, I assume you mean Zone ;)

Reply:
Thank you for noticing that